src/main/java/io/quarkus/search/app/fetching/FetchingService.java

```diff
@@ -60,42 +61,64 @@ public class FetchingService {
     public QuarkiverseIO fetchQuarkiverseIo(FailureCollector failureCollector) throws IOException {
         CloseableDirectory tempDir = CloseableDirectory.temp("quarkiverse-io");
         Path artifact = null;
-        if (quarkiverseIOConfig.githubArtifact().isPresent()) {
-            QuarkiverseIOConfig.GithubArtifact ghConfig = quarkiverseIOConfig.githubArtifact().get();
-            GitHub github = new GitHubBuilder().withOAuthToken(ghConfig.token()).build();
-            GHRepository repository = github.getRepository(ghConfig.repository());
-            for (GHWorkflowRun run : repository.queryWorkflowRuns()
-                    .conclusion(GHWorkflowRun.Conclusion.SUCCESS)
-                    .list().withPageSize(10)) {
-                if (ghConfig.actionName().equals(run.getName())) {
-                    for (GHArtifact ghArtifact : run.listArtifacts().withPageSize(5).toList()) {
-                        if (ghConfig.artifactName().equals(ghArtifact.getName())) {
-                            artifact = tempDir.path().resolve(ghArtifact.getName() + ".zip");
-                            final Path finalArtifact = artifact;
-                            Log.infof("Downloading Quarkiverse %s artifact #%s", ghConfig.artifactName(), ghArtifact.getId());
-                            ghArtifact.download(is -> Files.copy(is, finalArtifact));
-                            break;
-                        }
-                    }
-                    break;
+        switch (quarkiverseIOConfig.source()) {
+            case NONE -> {
+            }
+            case GITHUB_ARTIFACT -> {
+                if (quarkiverseIOConfig.githubArtifact().isPresent()) {
```

Comment:
```suggestion
```
we probably don't need that if statement anymore, no?

Reply:
Nice catch, fixed.